pennylane/qnn/keras.py

```diff
@@ -293,14 +286,14 @@ def call(self, inputs):
         Returns:
             tensor: output data
         """
-        if len(tf.shape(inputs)) > 1 and self.argnum is None:
+        if self.split_batches and len(tf.shape(inputs)) > 1:
```

Comment:
I'm wondering: Given that we do not support more than one parameter broadcasting axis, one could also iteratively unstack in case there are more than one axis, no?
Along the lines of 
```suggestion
        if ((ndim:=len(tf.shape(inputs))) > 1 and self.split_batches) or ndim>2:
```

Reply:
If we want to batch for more than one dimension, let's make the different cases very clear in the documentation because it will become hard for users to chose the right configuration.

Reply:
Oops, I think there's a much simpler solution by reshaping the arrays so that there's only one batch dimension, then reshaping them back at the end.

Reply:
At least the case of more than one broadcasting axis is solved. We will need the `split_batches` argument until we decide what to do with density matrices.